mmn/node_information.go

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2019 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package mmn
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"runtime"
+	"strings"
+
+	"github.com/mysteriumnetwork/go-ci/shell"
+	"github.com/pkg/errors"
+
+	"github.com/mysteriumnetwork/node/core/ip"
+	"github.com/mysteriumnetwork/node/metadata"
+)
+
+// NodeInformation contains node information to be sent to MMN
+type NodeInformation struct {
+	MACAddress  string `json:"mac_address"` // SHA256 hash
+	LocalIP     string `json:"local_ip"`
+	OS          string `json:"os"`
+	Arch        string `json:"arch"`
+	NodeVersion string `json:"node_version"`
+	Identity    string `json:"identity"`
+}
+
+// OnIdentityUnlockCallback sends node information to MMN on identity unlock
+func OnIdentityUnlockCallback(client *client, resolver ip.Resolver) func(string) {
```

Comment:
Function name does not say anything on what it does. 
```suggestion
func CollectNodeData(client *client, resolver ip.Resolver) func(string) {
```